pkg/util/validation/validate.go

```diff
@@ -68,51 +69,65 @@ const (
 	ExemplarMaxLabelSetLength = 128
 )
 
-// DiscardedSamples is a metric of the number of discarded samples, by reason.
-var DiscardedSamples = prometheus.NewCounterVec(
-	prometheus.CounterOpts{
-		Name: "cortex_discarded_samples_total",
-		Help: "The total number of samples that were discarded.",
-	},
-	[]string{discardReasonLabel, "user"},
-)
+type ValidateMetrics struct {
+	DiscardedSamples   *prometheus.CounterVec
+	DiscardedExemplars *prometheus.CounterVec
+	DiscardedMetadata  *prometheus.CounterVec
+}
 
-// DiscardedExemplars is a metric of the number of discarded exemplars, by reason.
-var DiscardedExemplars = prometheus.NewCounterVec(
-	prometheus.CounterOpts{
-		Name: "cortex_discarded_exemplars_total",
-		Help: "The total number of exemplars that were discarded.",
-	},
-	[]string{discardReasonLabel, "user"},
-)
+func registerCollector(r prometheus.Registerer, c prometheus.Collector) {
+	err := r.Register(c)
+	if err != nil && !errors.As(err, &prometheus.AlreadyRegisteredError{}) {
```

Comment:
Do we expect to have `prometheus.AlreadyRegisteredError` error?

Reply:
Yes, it will happen when the distributor and ingester are both instantiated in the test [TestCortex](https://github.com/cortexproject/cortex/blob/5742f57806dc8cec630828c2093443ceb9413416/pkg/cortex/cortex_test.go#L38)